lib/assertions.js

```diff
@@ -1796,27 +1796,35 @@ module.exports = expect => {
       if (valueType.is('array-like') && !subjectIsArrayLike) {
         expect.fail();
       }
-      const promiseByKey = {};
-      const keys = valueType.getKeys(value);
+
       const subjectKeys = subjectType.getKeys(subject);
+      const valueKeys = valueType.getKeys(value);
```

Comment:
Mhh I know this is my mistake, but I think things will be much more clear if we called subject actual and value expected in this method. We can always fix that later.

Reply:
or maybe subject and spec.